### PR TITLE
[TASK] #94243 - Send user session cookies as hash-signed JWT

### PR DESCRIPTION
**Pull request by brotkrueml**
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/217

### CODE REVIEW

File: Documentation/ApiOverview/Authentication/Index.rst [hunk 1]
```diff
@@ -84,6 +84,26 @@ by a login request. In the FE, this happens when a form field
...
+    `JSON Web Tokens (JWT) <https://jwt.io/>`__ are used to transport user
+    session identifiers in `be_typo_user` and `fe_typo_user` cookies.
+
+Using JWT's `HS256` (HMAC signed based on SHA256) allows to determine whether a
+session cookie is valid before comparing with server-side stored session data.
+This enhances the overall performance a bit, since sessions cookies would be
+checked for every request to TYPO3's backend and frontend.
+
+The session cookies can be pre-validated without querying the database, which
+can filter invalid requests and might reduce the enhances the overall
```
Comment (linawolf) [inline Documentation/ApiOverview/Authentication/Index.rst:97]: This sentence sounds bumpy

File: Documentation/ApiOverview/Authentication/Index.rst [hunk 2]
```diff
@@ -84,6 +84,26 @@ by a login request. In the FE, this happens when a form field
...
+performance a bit.
+
+As a consequence session tokens are not sent "as is", but are wrapped in a
```
Comment (linawolf) [inline Documentation/ApiOverview/Authentication/Index.rst:100]:
```suggestion
As a consequence session tokens are not sent "as are", but are wrapped in a
```

File: Documentation/ApiOverview/SessionStorageFramework/UserSessionManagement.rst [hunk 3]
```diff
@@ -97,10 +97,10 @@ The session object created or retrieved by the :php:`UserSessionManager` provide
 +---------------------+-------------+------------------------------------------------------------------------------+
 | Method              | Return type | Description                                                                  |
 +=====================+=============+==============================================================================+
-| getIdentifier()     | String      | Returns the session id. This is the :php:`ses_id` respectively the           |
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/UserSessionManagement.rst:100]: I would prefer to have this information in the source code and generate the API information automatically so it stays updated...

File: Documentation/ApiOverview/Authentication/Index.rst (hunk 2 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/Authentication/Index.rst:100]: Nope, "as is" should be right: https://dict.leo.org/englisch-deutsch/as%20is

File: Documentation/ApiOverview/SessionStorageFramework/UserSessionManagement.rst (hunk 3 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/SessionStorageFramework/UserSessionManagement.rst:100]: Yup, I agree and thought also about it, but it is not part of this PR.

File: Documentation/ApiOverview/Authentication/Index.rst (hunk 2 above)
Comment (linawolf) [reply Documentation/ApiOverview/Authentication/Index.rst:99]: It sounds wrong, I asked Tom as native speaker

Comment (linawolf) [reply Documentation/ApiOverview/Authentication/Index.rst:99]: Ok, Tom agrees with you

**Pull request merged**